frontend/src/components/form/email-input.jsx

```diff
@@ -0,0 +1,57 @@
+import { createFormControl } from "solid-forms";
+import { mergeProps, onMount, Show } from "solid-js";
+import { styled } from "solid-styled-components";
+import { updateValue } from "../../helpers";
+
+const EmailInput = (props) => {
+  const mergedProps = mergeProps({ control: createFormControl("") }, props);
+
+  onMount(() => {
+    mergedProps.control.setValue(
+      updateValue(mergedProps.name, mergedProps.control.value)
+    );
```

Comment:
I think we should put all stuff that works with IndexedDB and LocalStorage into the Train page component instead